### PR TITLE
[docs] Migrate Coding page as Policies index

### PR DESCRIPTION
**Pull request by sarjona**
Migrated pages:
- https://docs.moodle.org/dev/Coding

Apart from that, several pages have been marked as "WillNotMigrate". For instance:
- https://docs.moodle.org/dev/Accessibility (and several pages linked from there).
- https://docs.moodle.org/dev/XHTML

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/157"></a>

### CODE REVIEW

File: general/development/policies.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+title: Policies
+tags:
+  - Policies
+  - Coding guidelines
+  - Developer processes
+---
+
+:::note
+
+This page is the top-level page describing Moodle's coding guidelines.  It's the place to start if you want to know how to write code for Moodle.
```
Comment (andrewnicols) [inline general/development/policies.md:11]:
```suggestion
This is the top-level page describing Moodle's coding standards and guidelines.  It's the place to start if you want to know how to write code for Moodle.
```

File: general/development/policies.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+It's also a script running on people's servers, so Moodle needs to be a responsible Internet citizen and not introduce vulnerabilities that could allow crackers to gain unlawful access to the server it runs on.
+
+:::info
+
+Any single script (in Moodle core or a third party module) can introduce a vulnerability to thousands of sites, so it's important that all developers strictly follow our [Moodle security guidelines](/general/development/policies/security).
+
+:::
+
+## HTML5 and CSS
```
Comment (andrewnicols) [inline general/development/policies.md:59]: Consider renaming this since it's not just HTML5 and CSS, but also WCAG, ARIA, etc.

```suggestion
## Standards
```

File: general/development/policies.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+It's important that Moodle produces strict, well-formed [HTML 5](http://en.wikipedia.org/wiki/HTML5) code (preferably backwards compatible with XHTML 1.1 if possible), compliant with all common accessibility guidelines (such as [W3C WAG 2.0](http://www.w3.org/TR/WCAG20/), [ARIA](http://www.w3.org/TR/wai-aria-practices/)).
```
Comment (andrewnicols) [inline general/development/policies.md:61]:
```suggestion
It's important that Moodle produces strict, well-formed [HTML 5](http://en.wikipedia.org/wiki/HTML5) code (preferably backwards compatible with XHTML 1.1 if possible), compliant with all common accessibility guidelines (such as [W3C WCAG 2.0](http://www.w3.org/TR/WCAG20/), [ARIA](http://www.w3.org/TR/wai-aria-practices/)).
```

File: general/development/policies.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+CSS should be used for layout. Moodle comes with several themes installed. Beginning with version 2.7, only the 'Clean' theme comes in the base Moodle code. The 'standard' theme, which should be a plain theme suitable to act as a building block for other themes. That should contain the minimal styling to make Moodle look OK and be functional. Then Moodle comes with several other default themes that look good and demonstrate various techniques for building themes.
```
Comment (andrewnicols) [inline general/development/policies.md:63]: THis needs updating for Boost/Classic vs. Clean + Standard.

```suggestion
CSS should be used for layout. Moodle comes with several themes installed. Beginning with version 2.7, only the 'Boost, and 'Classic' themes are included with the base Moodle code.

:::note Writing your own theme

We recommend that if you are writing your own theme that it should extend the Moodle 'Boost' theme.

:::
```

The rest seems to be irrelevant now I think.

File: general/development/policies.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+:::info
+
+For more information and advice, see [Performance and scalability](https://docs.moodle.org/dev/Performance_and_scalability).
+
+:::
+
+## Database
+
+Moodle has a powerful database abstraction layer that we wrote ourselves, called [XMLDB](https://docs.moodle.org/dev/XMLDB_Documentation).  This lets the same Moodle code work on MySQL/MariaDB, PostgreSQL, MS SQL Server and Oracle. There are know issues when using Oracle, it is not fully supported and is not recommended for production sites.
```
Comment (andrewnicols) [inline general/development/policies.md:129]:
```suggestion
Moodle has a powerful database abstraction layer that we wrote ourselves, called [XMLDB](https://docs.moodle.org/dev/XMLDB_Documentation).  This lets the same Moodle code work on MySQL/MariaDB, PostgreSQL, MS SQL Server and Oracle. There are known issues when using Oracle, it is not fully supported and is not recommended for production sites.
``

Is this Oracle statement true?

File: general/development/policies.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+We have tools and API for [defining and modifying tables](https://docs.moodle.org/dev/DDL_functions) as well as [methods for getting data in and out](https://docs.moodle.org/dev/Data_manipulation_API) of the database.
```
Comment (andrewnicols) [inline general/development/policies.md:131]:
```suggestion
We have tools and APIs for [defining and modifying tables](https://docs.moodle.org/dev/DDL_functions), and [retrieving and storing data](https://docs.moodle.org/dev/Data_manipulation_API) in the database.
```

File: general/development/policies.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+:::info
+
+For more about this, see [Moodle Database](https://docs.moodle.org/dev/Database) guidelines.
```
Comment (andrewnicols) [inline general/development/policies.md:135]:
```suggestion
For more about this, see the [Moodle Database](https://docs.moodle.org/dev/Database) guidelines.
```

File: general/development/policies.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+:::
+
+## Events
+
+Moodle allows inter-module communication via **events**.  Modules can **trigger** specific events and other modules can choose to **handle/observe** those events.
```
Comment (andrewnicols) [inline general/development/policies.md:141]: Err... this is not true.https://deploy-preview-157--peppy-zabaione-65df4a.netlify.app/general/development/policies/component-communication#event-observers

> The normal rules for inter-component communication apply.

**Review by andrewnicols — DISMISSED** [review]
Mostly minor changes but some content changes to fix out-of-date information.

I'd also suggest that we change the 'Policies' header navbar link to point to the new page too. That's configured in `config/navbar.js`

File: general/development/policies.md (hunk 8 above)
Comment (sarjona) [reply general/development/policies.md]: You're right! :-)  I've reviewed it :-)

**Comment by sarjona** [comment]
Hi @andrewnicols!
Thanks for your suggestions :-) I've fixed them all (but the link you mentioned because, as you probably saw, the Coding link in the navbar now points to the Quick start page. I think that's OK for you :-)
The patch is ready again to get your eyes over it ;-)
